WordPress-VIP-Go/ruleset-test.inc

```diff
@@ -0,0 +1,330 @@
+is_writeable( $file ); // Error.
+
+// WordPressVIPMinimum.Functions.RestrictedFunctions.file_ops_link
+link( 'file.txt', 'newfile.txt' ); // Error.
+
+// WordPressVIPMinimum.Functions.RestrictedFunctions.file_ops_rename
+rename( 'oldfile.txt', $file ); // Error.
+
+// WordPressVIPMinimum.Functions.RestrictedFunctions.file_ops_symlink
+symlink( $file, 'file.txt' ); // Error.
+
+// WordPressVIPMinimum.Functions.RestrictedFunctions.file_ops_tempname
+tempnam( $dir, 'pre' ); // Error.
+
+// WordPressVIPMinimum.Functions.RestrictedFunctions.file_ops_touch
+touch( $file ); // Error.
+
+// WordPressVIPMinimum.Functions.RestrictedFunctions.file_ops_unlink
+unlink( $file ); // Error.
+
+// WordPressVIPMinimum.Functions.RestrictedFunctions.cookies_setcookie
+setcookie( 'cookie[three]', 'cookiethree' ); // Error. 
+
+// WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___COOKIE
+$x = isset( $_COOKIE['bar'] ) ? sanitize_key( $_COOKIE['bar'] ) : ''; // Error.
+
+// WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
+if ( ! isset( $_SERVER['HTTP_USER_AGENT'] ) ) { // Error.
+}
+
+// Generic.NamingConventions.ConstructorName.OldStyle
+class OldClass
+{
+    function OldClass() // Error.
+    {
+    }
+}
+
+// WordPress.Security.EscapeOutput.OutputNotEscaped
+echo $foo; // Error.
+echo 'Some Raw String';  // Ok.
+echo esc_html( $good . $better ) . $foo; // Error.
+echo esc_html( $food . 'include'  ); // Ok.
+echo esc_html( strtoupper( $ok ) ) . ' ' . esc_html( strtolower( $ok ) ); // Ok.
+echo filter_var( $bar, FILTER_VALIDATE_EMAIL ); // Ok.
+echo filter_input( INPUT_GET, $bar, FILTER_SANITIZE_SPECIAL_CHARS ); // Ok.
+echo ent2ncr( $text ); // Error.
+echo empty( $foo ) ? 'no foo' : $foo; // Error.
+echo $foo ? esc_html( $foo ) : 'no foo'; // Ok.
+
+// WordPress.WP.AlternativeFunctions.file_system_read_fclose
+fclose( $fp ); // Error.
+
+// WordPress.WP.AlternativeFunctions.file_system_read_fopen
+fopen( 'file.txt', 'r' ); // Error.
+
+// WordPressVIPMinimum.Performance.FetchingRemoteData.FileGetContentsUnknown
+$external_resource = file_get_contents( $test ); // Error.
+$file_content = file_get_contents( 'my-file.svg' ); // Ok.
+wpcom_vip_file_get_contents( $bar ); // Ok.
+
+// WordPress.Security.NonceVerification.NoNonceVerification
+function bar() {
+	if ( ! isset( $_POST['test'] ) ) { // Warning.
+		return;
+	}
+}
+function foo_bar() {
+	if ( ! isset( $_POST['test'] ) || ! wp_verify_nonce( 'some_action' ) ) { // Ok.
+		exit;
+	}
+}
+
+// WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+do_something( $_POST ); // Ok.
+if ( isset( $_POST['foo2'] ) ) {
+	bar( wp_unslash( $_POST['foo2'] ) ); // Warning.
+}
+
+// WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
+$args = array(
+	'posts_per_page' => 999, // Warning.
+	'posts_per_page' => 1, // OK.
+	'posts_per_page' => '1', // OK.
+);
+_query_posts( 'posts_per_page=999' ); // Warning.
```

Comment:
Why is this one a warning - this appears to be a custom function `_query_posts()`.